### PR TITLE
Fix Typescript errors and setup Vite

### LINKED ISSUE #1: Move example to Vite
Right now its using parcel and I have come into some issues of it not recognizing changes

https://stackblitz.com/edit/vitejs-vite-61vmy4?file=index.html&terminal=dev

### PR DESCRIPTION
**Pull request by jfranciscosousa**
Closes #1 and #2 

Fixes TS errors. `next/router` import needed a `default` there and setup `React.lazy` typings.

Also migrates the example to `vite` instead of `parcel`.

### CODE REVIEW

**Comment by SaraVieira** [comment]
Hero!!

File: src/index.tsx [hunk 1]
```diff
@@ -66,6 +65,7 @@ export const closeModal = () => {
   urlParams.delete(MODAL_KEY);
 
   routerPush(createURL(urlParams));
+  console.log(window.location.search)
```
Comment (SaraVieira) [inline src/index.tsx:68]: Shakes head

**Comment by SaraVieira** [comment]
Some co sole logs were left out but other than that looks awesome!!

**Comment by SaraVieira** [comment]
Removed the logs and pulled from master

Vite workls perfect and reloads
Its like magic

Merge whenever

**Comment by jfranciscosousa** [comment]
Awesome, thanks!

**Pull request merged**